File: Arrays/Python/Maximum_Index.py
class Solution:
    #Complete this function
    # Function to find the maximum index difference.
    def maxIndexDiff(self,A, N): 
        ##Your code here
        low = 0
        high = N-1
        maxi = -1
        while low <= high:
            if A[low] <= A[high]: 
                maxi = max(maxi, high-low)
                high = N-1
                low += 1
            else:
                high -= 1
        return maxi

File: Solutions/Java/Maximum_Index.java
class Solution{
    
    // A[]: input array
    // N: size of array
    // Function to find the maximum index difference.
    static int maxIndexDiff(int A[], int N) { 
        
        // Your code here
        int low = 0;
        int high = N-1;
        int max = -1;
        while (low <= high) {
            if (A[low] <= A[high]) { 
                max = Math.max(max,high-low);
                high = N-1;
                low++;
            }
            else {
                high--;
            }
        }
        return max; 
    }
}
